modules/hostap/Kconfig

```diff
@@ -8,8 +8,9 @@
 menuconfig WPA_SUPP
 	bool "WPA supplicant support"
 	# Need full POSIX from libc, Zephyr's POSIX support is only partial
-	depends on !POSIX_API
 	select POSIX_CLOCK
+	select POSIX_SIGNAL
```

Comment:
Can this be avoided? In Matter we use `NET_SOCKETS_POSIX_NAMES` which seems mutually exclusive with POSIX_API.

Reply:
This is the reason why I always try to use `zsock_` calls and leave the `POSIX_NAMES` or `POSIX_API` for application to choose.

This is constant source of conflicts.

Reply:
Here's some reference: https://github.com/zephyrproject-rtos/zephyr/issues/56294

But to answer your question - I didn't really find an alternative to mitigate the issues with POSIX signal API usage, other than finally enabling the POSIX_SIGNAL (and POSIX_API as a consequence) for the supplicant. The recent reworks upstream made the libc headers/definitions incompatible with rest, causing build errors.

As for `NET_SOCKETS_POSIX_NAMES` - the transition to `POSIX_API` isn't that bad, just need to add a few extra headers, see other commits like the one with `https_client`. Or just stick to `zsock_` as suggested.

Reply:
> I think it's quite unfortunate that a low-level component forces POSIX while this should be an application's decision, but if it's unavoidable we will have to deal with it.

Technically WPA supplicant is an application :) which was turned in to a library.